python/cuda_parallel/tests/test_reduce_api.py

```diff
@@ -37,3 +37,34 @@ def min_op(a, b):
     expected_output = 0
     assert (d_output == expected_output).all()
     # example-end reduce-min
+
+
+def test_reduce_struct_type():
+    from cuda.parallel.experimental.struct import gpu_struct
+
+    # example-begin reduce-struct
+    @gpu_struct
+    class Pixel:
+        r: np.int32
+        g: np.int32
+        b: np.int32
+
+    def max_g_value(x, y):
+        return x if x.g > y.g else y
+
+    d_rgb = cp.random.randint(0, 256, (10, 3), dtype=np.int32).view(Pixel.dtype)
+    d_out = cp.zeros(1, Pixel.dtype)
+
+    h_init = Pixel(0, 0, 0)
+
+    reduce_into = algorithms.reduce_into(d_rgb, d_out, max_g_value, h_init)
+    temp_storage_bytes = reduce_into(None, d_rgb, d_out, len(d_rgb), h_init)
+
+    d_temp_storage = cp.zeros(temp_storage_bytes, dtype=np.uint8)
```

Comment:
forgot to ask:
1. why bother zero-init'ing the temp storage? 
2. I should already know this but I can't recall... does `reduce_info` get a stream from `d_out` & CI through CAI for kernel launch?

Reply:
1. Changed it to empty.
2. Not yet (https://github.com/NVIDIA/cccl/issues/3080)

Reply:
Got it so I assume everything is running on CuPy current (null) stream for now.